### PR TITLE
Update mimir-prom to v1.8.2-0.20250423083340-007de9b763aa

### PR DESCRIPTION
**Pull request by zenador**
#### What this PR does

This updates mimir-prom to https://github.com/grafana/mimir-prometheus/pull/867

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by dimitarvdimitrov — COMMENTED** [review]
do you know if we now support duration arithmetic? Do we not support it explicitly?

File: pkg/streamingpromql/engine_test.go [hunk 1]
```diff
@@ -356,13 +356,6 @@ func TestRangeVectorSelectors(t *testing.T) {
 				},
 			},
 		},
-		"floats: 0 length range": {
```
[the remark this answers is not in the document]
Comment (56quarters) [reply pkg/streamingpromql/engine_test.go:359]: @charleskorn added these because he found an edge case in promql while working on MQE and added them to Prometheus upstream and MQE. The equivalent tests upstream (Prometheus) were removed when duration arithmetic support was added which we are now pulling in.

**Comment by 56quarters** [comment]
> do you know if we now support duration arithmetic? Do we not support it explicitly?

There's a global flag that needs to be set to enable it (same as experimental functions). We don't set it currently and I didn't want to just blindly enable it during a Prometheus update without talking with the team. I _think_ we don't need to do anything to support it since it's entirely in the upstream PromQL parser which we use even in MQE.

**Comment by zenador** [comment]
@krajorama asked about that in slack (not sure if I should stick the link here), it seems we're planning to enable it by default for now, but we were planning to do that in a separate PR with a proper changelog as it's cleaner. Right now it's not enabled, but some of the validation behaviour has changed anyway (returning different errors, tests have been updated accordingly).

File: pkg/streamingpromql/engine_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zenador) [reply pkg/streamingpromql/engine_test.go:359]: Zero durations are no longer allowed in PromQL with https://github.com/prometheus/prometheus/pull/16249

File: pkg/frontend/querymiddleware/results_cache.go [hunk 2]
```diff
@@ -324,7 +324,12 @@ func areEvaluationTimeModifiersCachable(r MetricsQueryRequest, maxCacheTime int6
 	}
 
 	// This resolves the start() and end() used with the @ modifier.
-	expr = promql.PreprocessExpr(expr, timestamp.Time(r.GetStart()), timestamp.Time(r.GetEnd()))
+	expr, err = promql.PreprocessExpr(expr, timestamp.Time(r.GetStart()), timestamp.Time(r.GetEnd()))
+	if err != nil {
+		// We are being pessimistic in such cases.
+		level.Warn(logger).Log("msg", "failed to preprocess query, considering @ modifier as not cachable", "query", query, "err", err)
```
[the remark this answers is not in the document]
Comment (zenador) [reply pkg/frontend/querymiddleware/results_cache.go:330]: Isn't that the same behaviour (different log msg) we already had in this though?

```
	expr, err := parser.ParseExpr(query)
	if err != nil {
		// We are being pessimistic in such cases.
		level.Warn(logger).Log("msg", "failed to parse query, considering @ modifier as not cachable", "query", query, "err", err)
		return false
	}
```

Or should we change both of those to return unique errors and remove the log line?

[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply pkg/frontend/querymiddleware/results_cache.go:330]: ah, i didn't see this before. I'd prefer if we return a reason so we don't double log the same error. But i understand if you consider this out of scope

**Review by dimitarvdimitrov — APPROVED** [review]
thanks for explaining @56quarters @zenador

**Comment by dimitarvdimitrov** [comment]
> > do you know if we now support duration arithmetic? Do we not support it explicitly?
> 
> There's a global flag that needs to be set to enable it (same as experimental functions). We don't set it currently and I didn't want to just blindly enable it during a Prometheus update without talking with the team. I _think_ we don't need to do anything to support it since it's entirely in the upstream PromQL parser which we use even in MQE.

I was more concerned about Mimir crashing or panicking if its gets a request with arithmetic

File: pkg/frontend/querymiddleware/results_cache.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (zenador) [reply pkg/frontend/querymiddleware/results_cache.go:330]: Happy to fix that in a separate PR!

[the remark this answers is not in the document]
Comment (zenador) [reply pkg/frontend/querymiddleware/results_cache.go:330]: https://github.com/grafana/mimir/pull/11312
